src/modules/auth/viewModels/login-data.ts

```diff
@@ -1,4 +1,25 @@
 export class LoginData {
+  isValid = false;
   email = '';
   password = '';
+
+  rules = {
+    email: [
+      (v: string) => !!v || 'E-mail is required',
+      (v: string) => /.+@.+\..+/.test(v) || 'E-mail must be valid',
+    ],
+    password: [
+      (v: string) => !!v || 'Password is required',
+      (v: string) =>
+        v.length >= 8 || 'Password must be at least 8 characters',
+    ],
+  };
+
+  validate(): void {
+    this.isValid =
+      this.rules.email.every((rule) => rule(this.email) === true) &&
+      this.rules.password.every(
+        (rule) => rule(this.password) === true,
+      );
+  }
 }
```

Comment:
To ciekawe podejście, a nie myślałeś czasem aby wykorzystać do tego gotowe rozwiązania typu Zod lub Yup?

Reply:
Testowałem to rozwiązanie bo musiałem poprawić walidację:D Będę używał Zod do bardziej rozbudowanych formów w tym projekcie.